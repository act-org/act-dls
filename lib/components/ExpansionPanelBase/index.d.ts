/**
 * @prettier
 */
import * as React from 'react';
import { ExpansionPanelProps } from '@material-ui/core';
export declare type Props = ExpansionPanelProps;
declare const ExpansionPanelBase: React.FC<Props>;
export default ExpansionPanelBase;
//# sourceMappingURL=index.d.ts.map