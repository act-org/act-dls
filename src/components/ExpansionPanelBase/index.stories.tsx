/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean } from '@storybook/addon-knobs';

import ExpansionPanelBase from '.';

export const Base = (): React.ReactElement<any> => (
  <ExpansionPanelBase
    expanded={boolean('Expanded?', false)}
    onChange={action('onChange()')}
    square={boolean('Square?', false)}
  >
    <div>ExpansionPanel</div>
  </ExpansionPanelBase>
);

export default {
  component: ExpansionPanelBase,
  parameters: {
    info: {
      text: 'This is the base variant of the ExpansionPanel component.',
    },
  },
  title: 'Atoms/ExpansionPanel',
};
