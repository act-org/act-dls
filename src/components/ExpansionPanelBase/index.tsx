/**
 * @prettier
 */

import * as React from 'react';
import { ExpansionPanel, ExpansionPanelProps } from '@material-ui/core';

export type Props = ExpansionPanelProps;

const ExpansionPanelBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <ExpansionPanel {...props} />;

export default ExpansionPanelBase;
