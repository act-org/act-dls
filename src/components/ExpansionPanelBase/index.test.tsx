/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import ExpansionPanelBase from '.';

describe('ExpansionPanelBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <ExpansionPanelBase defaultExpanded expanded onChange={noop} square>
        <div>ExpansionPanelBase</div>
      </ExpansionPanelBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
