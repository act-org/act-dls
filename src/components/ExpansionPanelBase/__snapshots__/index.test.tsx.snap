// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpansionPanelBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiExpansionPanel-root Mui-expanded MuiPaper-elevation1"
  >
    <div>
      ExpansionPanelBase
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
